test/blackbox-tests/test-cases/no-installable-mode/run.t

```diff
@@ -0,0 +1,19 @@
+When a public executable is built in shared_object mode, a specific error
+message is displayed:
+
+  $ dune build --root=public --display=short
+  File "jbuild", line 4, characters 2-74:
+  Error: No installable mode found for this element.
+  One of the following modes is required:
+   - (best exe)
```

Comment:
Actually, could you change `Link_mode.sexp_of_t` so that it prints the short version here? As a user, I would copy and paste what the tool suggests, so I think it's better to use the nicer representation.